python/foglamp/services/core/api/service.py

```diff
@@ -143,21 +153,44 @@ def get_service_records():
+            # Revert scheduled_process entry
+            payload = PayloadBuilder().WHERE(['name', '=', name]).payload()
+            result = storage.delete_from_tbl('scheduled_processes', payload)
             raise web.HTTPInternalServerError(reason='Plugin "{}" import problem from path "{}". {}'.format(plugin, plugin_module_path, str(ex)))
         except Exception as ex:
+            # Revert scheduled_process entry
+            payload = PayloadBuilder().WHERE(['name', '=', name]).payload()
+            result = storage.delete_from_tbl('scheduled_processes', payload)
             raise web.HTTPInternalServerError(reason='Failed to create plugin configuration. {}'.format(str(ex)))
 
-        # Next add a schedule to run the new service at startup
-        schedule = StartUpSchedule()  # TODO: For North plugin also?
-        schedule.name = name
-        schedule.process_name = name
-        schedule.repeat = datetime.timedelta(0)
-        schedule.exclusive = True
-        schedule.enabled = False  # if "enabled" is supplied, it gets activated in save_schedule() via is_enabled flag
-
-        # Save schedule
-        await server.Server.scheduler.save_schedule(schedule, is_enabled)
-        schedule = await server.Server.scheduler.get_schedule_by_name(name)
+        # Lastly add a schedule to run the new service at startup
+        try:
+            schedule = StartUpSchedule()  # TODO: For North plugin also?
+            schedule.name = name
+            schedule.process_name = name
+            schedule.repeat = datetime.timedelta(0)
+            schedule.exclusive = True
+            schedule.enabled = False  # if "enabled" is supplied, it gets activated in save_schedule() via is_enabled flag
+
+            # Save schedule
+            await server.Server.scheduler.save_schedule(schedule, is_enabled)
+            schedule = await server.Server.scheduler.get_schedule_by_name(name)
+        except StorageServerError as ex:
+            # Revert configuration entry
+            payload = PayloadBuilder().WHERE(['key', '=', name]).payload()
+            result = storage.delete_from_tbl('configuration', payload)
+            # Revert scheduled_process entry
+            payload = PayloadBuilder().WHERE(['name', '=', name]).payload()
+            result = storage.delete_from_tbl('scheduled_processes', payload)
+            raise web.HTTPInternalServerError(reason='Failed to created schedule. {}'.format(ex.error))
+        except Exception as ins_ex:
+            # Revert configuration entry
+            payload = PayloadBuilder().WHERE(['key', '=', name]).payload()
+            result = storage.delete_from_tbl('configuration', payload)
+            # Revert scheduled_process entry
+            payload = PayloadBuilder().WHERE(['name', '=', name]).payload()
+            result = storage.delete_from_tbl('scheduled_processes', payload)
+            raise web.HTTPInternalServerError(reason='Failed to created schedule. {}'.format(str(ins_ex)))
```

Comment:
repeated def for ` # Revert configuration entry` can we have it as a def please?